fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1512,432 +1526,477 @@ private Bundle processBundleEntries(Bundle requestBundle, Bundle responseBundle,
-            // This list will contain the indices associated with only the entries for the specified http method.
-            List<Integer> entryIndices =
-                    getBundleRequestIndicesForMethod(requestBundle, responseBundle, httpMethod);
-            if (log.isLoggable(Level.FINER)) {
-                log.finer("Bundle request indices to be processed: " + entryIndices.toString());
-            }
-
-            // Next, for PUT (update) requests, extract any local identifiers and resolve them ahead of time.
-            // We do this to prevent any local reference problems from occurring due to our re-ordering
-            // of the PUT request entries.
-            if (httpMethod.equals(HTTPVerb.PUT)) {
-                if (log.isLoggable(Level.FINER)) {
-                    log.finer("Pre-processing bundle request entries for PUT method...");
-                }
-                for (Integer index : entryIndices) {
-                    Bundle.Entry requestEntry = requestBundle.getEntry().get(index);
-
-                    // Retrieve the local identifier from the request entry (if present).
-                    String localIdentifier = retrieveLocalIdentifier(requestEntry, localRefMap);
-
-                    // Since this is for a PUT request (update) we should be able to resolve the local identifier
-                    // prior to processing the request since the resource's id must already be contained in the resource
-                    // within the request entry.
-                    if (localIdentifier != null) {
-                        Resource resource = requestEntry.getResource();
-                        addLocalRefMapping(localRefMap, localIdentifier, resource);
-                    }
-                }
-            }
-
-            // Next, for PUT and DELETE requests, we need to sort the indices by the request url path value.
-            if (httpMethod.equals(HTTPVerb.PUT) || httpMethod.equals(HTTPVerb.DELETE)) {
-                sortBundleRequestEntries(requestBundle, entryIndices);
-                if (log.isLoggable(Level.FINER)) {
-                    log.finer("Sorted bundle request indices to be processed: "
-                            + entryIndices.toString());
-                }
-            }
-
-            // Now visit each of the request entries using the list of indices obtained above.
-            // Use hashmap to store both the index and the according updated response bundle entry.
-            HashMap<Integer, Bundle.Entry> responseIndexAndEntries =
-                    new HashMap<Integer, Bundle.Entry>();
-            for (Integer entryIndex : entryIndices) {
+            // Visit each of the request entries, processing those with the specified request method.
+            // Use hashmap to store both the index and the accordingly updated response bundle entry.
+            HashMap<Integer, Bundle.Entry> responseIndexAndEntries = new HashMap<Integer, Bundle.Entry>();
```

Comment:
```suggestion
            Map<Integer, Bundle.Entry> responseIndexAndEntries = new HashMap<Integer, Bundle.Entry>();
```
Shouldn't the interface be sufficient here?

Reply:
yes, fixed